ADR/0039-creating-pending-pipelineruns.md

```diff
@@ -0,0 +1,66 @@
+# 39. Creating pending pipeline runs
+
+Date: 2024-11-21
+
+## Status
+
+Accepted
+
+## Context
+
+There is a frequent use case amongst Konflux users when triggering large amount of builds: sometimes users do not care how quickly the pipeline run will finish, but they want it to finish eventually.
+Now what happens is that part of the pipeline runs started in this big bulk is blocked by unavailable resources (e.g. namespace memory limit quotas or MPC pool size).
+Because of how Tekton works, no matter what these pipeline runs are considered running (even if their pods are waiting for resources) and so eventually timeout setting strikes.
+
+So say you started 300 pipeline runs at Fry evening and you expect to see 300 new images at Mon morning.
+You have resource quota for 20 concurrent builds, so 20 builds get started, when they finish 20 more and so on.
+But because all 300 pipeline runs started (from Tekton point of view) when they were created, after few iterations all the remaining pipeline runs will be cancelled (as timeouted).
```

Comment:
```suggestion
But because all 300 pipeline runs started (from Tekton point of view) when they were created, after few iterations all the remaining pipeline runs will be cancelled (due to a timeout).
```